client/gutenberg/extensions/slideshow/edit.js

```diff
@@ -74,12 +75,15 @@ class SlideshowEdit extends Component {
 			filesList: files,
 			onFileChange: images => {
 				const imagesNormalized = images.map( image => pickRelevantMediaFiles( image ) );
-				setAttributes( {
-					images: [ ...imagesNormalized, ...currentImages ],
-				} );
 				if ( ! imagesNormalized.every( image => isBlobURL( image.url ) ) ) {
 					unlockPostSaving( lockName );
+					this.setState( { jumpToIndex: null } );
+				} else {
+					this.setState( { jumpToIndex: currentImages.length } );
```

Comment:
Hmm can we do without the additional `state`? (And `Slideshow` prop?) `jumpToIndex` seems super implementation detail-ish -- can we derive it from `prevProps.images` and `this.props.images` in `Slideshow`'s `componentDidUpdate` method?

Reply:
Hmmm, so your idea would be that any time there are new images we jump to the end of the previous group + 1?

Reply:
Yeah -- does that make sense? The difference in functionality from your approach would be in that it'd also do that when images are added from the media library (rather than uploaded), right? Sounds reasonable too tho?